File: mini-package-api/mini-package-api/Models/Address.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.IO;
using System.Reflection.Emit;
using System.Text.Json.Serialization;

namespace mini_package_api.Models
{
    public class Address
    {
        public int AddressId { get; set; }

        public int CountryId { get; set; }

        public virtual Country Country { get; set; }

        public string City { get; set; }

        public string Street { get; set; }

        public string StreetNumber { get; set; }  

        public int? FlatNumber { get; set; }

        public string ZipCode { get; set; }

        public virtual List<ClientAddress> ClientAddresses { get; set; }
    }
}

File: mini-package-gui/src/app/models/address.ts
import { Country } from "./country";

export class Address {
  addressId: number;
  city: string;
  street: string;
  streetNumber: string;
  flatNumber: number | null;
  zipCode: string;
  country: Country | null;

  constructor() {
    this.addressId = 0;
    this.city = '';
    this.street = '';
    this.streetNumber = '';
    this.flatNumber = null;
    this.zipCode ='';
    this.country = null;
  }

  public get stringAddress() {
    let countryName = "";
    if (this.country != null && this.country != undefined)
      countryName = this.country.countryName + ' ';
    let flatNumber = "";
    if (this.flatNumber != null && this.flatNumber != undefined)
      flatNumber = '/' + this.flatNumber.toString();
    return `${countryName}${this.city} ${this.street} ${this.streetNumber}${flatNumber} ${this.zipCode}`
  }
}
